### PR TITLE
Build artifacts only for the used CPU architectures

### PR DESCRIPTION
**Pull request by hakman**
This PR reverts the previously added behaviour that was adding both AMD64 and ARM64 assets to each IG user data. The main reason for it is the increase in the size of the user data, which generated issues for users like #10202.

The CPU architecture of each IG can be changed using a new IG field named `Architecture`.
https://github.com/kubernetes/kops/blob/43b9d253ae09c757d4f0c0606385776bd114ccf6/pkg/apis/kops/instancegroup.go#L93-L94

It defaults to AMD64, but there is also a simple check for `-arm64` in the image name that should cover most cases for auto-detecting ARM64:
https://github.com/kubernetes/kops/blob/599fb7d993cf4355f8a5ef9aad5b9472b80641d9/upup/pkg/fi/cloudup/apply_cluster.go#L167-L174

/cc @justinsb @johngmyers @olemarkus @rdrgmnzs

### CODE REVIEW

**Comment by olemarkus** [comment]
It feels like there are better ways to optimise the template files. E.g zipping + base64 encode the strings yield 25-50% fewer bytes depending on the length of the spec.

File: upup/pkg/fi/cloudup/apply_cluster.go [hunk 1]
```diff
@@ -163,6 +163,24 @@ func (c *ApplyClusterCmd) Run(ctx context.Context) error {
 		}
 	}
 
+	for _, ig := range c.InstanceGroups {
+		if ig.Spec.Architecture == "" {
+			// TODO: Find CPU arch by using the image/instance info
+			if strings.Contains(ig.Spec.Image, "-arm64") {
+				ig.Spec.Architecture = string(architectures.ArchitectureAmd64)
```
Comment (rdrgmnzs) [inline upup/pkg/fi/cloudup/apply_cluster.go:170]: Did you mean ArchitectureArm64 here?

**Review by rdrgmnzs — COMMENTED** [review]
LGTM, besides the one question I have. Thanks @hakman!

IMO including the hashes of binaries that will never be used doesn't make that much sense given the user data size constraints.

**Comment by rdrgmnzs** [comment]
One issue I just hit while testing out gziping the user-data is that facter (puppet) does not handle gzip user-data well https://tickets.puppetlabs.com/browse/FACT-1354, so it may cause issues for any users (like my self) that have puppet installed on their hosts.

**Comment by olemarkus** [comment]
Yep. Hence the gzip+base64 of just those heredoc blocks.

**Comment by rdrgmnzs** [comment]
Ah, sorry I misunderstood then.

**Comment by hakman** [comment]
Closing in favor of #10357

**Pull request closed without merging**